rewriter/flatten.cc

```diff
@@ -352,7 +379,12 @@ class FlattenWalk {
         // if we get a MethodData back, then we need to move this and replace it
         if (auto md = methods.popScope()) {
             // we'll replace MethodDefs with the symbol that corresponds to the name of the method
-            auto replacement = ast::MK::Symbol(methodDef->loc, methodDef->name);
+            unique_ptr<ast::Expression> replacement;
+            if (methods.atClassTopLevel()) {
+                replacement = ast::MK::EmptyTree();
```

Comment:
Let's extract this to a separate PR.
I have FUD about this. Let's either be systematic and always use symbol or always use empty tree